### ISSUE TITLE
Building for cordova-ios 4.0.0 and wkWebView beta fails

### ISSUE THREAD

**Issue by runspired** [MobileChromeApps/cordova-plugin-service-worker#15]
```
/Plugins/cordova-plugin-service-worker/CDVServiceWorker.m:141:19: warning: 
      unused variable 'options' [-Wunused-variable]
    NSDictionary *options = [command argumentAtIndex:1];
                  ^
/Plugins/cordova-plugin-service-worker/CDVServiceWorker.m:285:32: error: 
      no known class method for selector 'dataFromBase64String:'
        NSData *data = [NSData dataFromBase64String:[response[@"body"] toString]];
                               ^~~~~~~~~~~~~~~~~~~~
```

**Comment by jamessharp** [comment]
[This code](https://github.com/ortoo/cordova-plugin-service-worker/commit/2ea1c846d6f60d32860d1795dc0a9475204bdcc2) fixes the build, however WKWebView doesn't allow interception of network requests which means the plugin doesn't (can't ?) work anyway.

**Comment by runspired** [comment]
@jamessharp hrm, so it's still goof for a lot of the uses of a service worker (data sync), but not for asset cacheing.  I'm ok with that personally since on cordova I'm serving from local assets, but would be nice if this was possible.  Does the cordova-whitelist plugin not intercept?

**Comment by jamessharp** [comment]
Yes, I suspect everything but asset cacheing/interception will work OK (I haven't tested it though). Chrome on iOS is still using UIWebView because of this issue (among other things). See [this meta-bug](https://code.google.com/p/chromium/issues/detail?id=423444) for discussion. You could hack round it for XHR requests by using something like [xhook](https://github.com/jpillora/xhook) but I don't think it's possible for standard requests (images, css, fonts etc).

**Comment by clelland** [comment]
@jamessharp That's right, there's no way to intercept network requests with WKWebView; this plugin won't work with cordova-plugin-wkwebview-engine, at least for fetch events.

@runspired, I don't believe that cordova-whitelist works on WKWebView either; the suggestion there is to use CSP to control what the web content of your application has access to

**Comment by runspired** [comment]
If you haven't already, you should open issue tickets with Apple for both ServiceWorker support (I've already done so) and for wkWebView to support intercepting network requests (I'm about to do so).

**Comment by runspired** [comment]
@jamessharp @clelland I've often wondered if you could "hook" the `src` attribute of `img` and `script` programmatically, I'll investigate that.  Alternative idea, a script included before anything else that grabs `src` and `link` attributes from `script` `img` and `link` and removes them, then adds a path from cache returned by the service worker.
